### PR TITLE
Add new microbenchmarks for query shapes: On Conflict, Update improvements, Select for updates, Backward scan post updates

### PR DESCRIPTION
**Pull request by shantanugupta-yb**
Reports for: 

- On Conflict: https://perf.dev.yugabyte.com/report/view/W3sibmFtZSI6InRlc3QtaWQiLCJpc0Jhc2VsaW5lIjp0cnVlLCJ0ZXN0X2lkIjo2ODI1MzAyfV0=
- Update improvements:  https://perf.dev.yugabyte.com/report/view/W3sibmFtZSI6InRlc3QtaWQiLCJpc0Jhc2VsaW5lIjp0cnVlLCJ0ZXN0X2lkIjo2ODU3MDAyfV0=
- Select for updates:  https://perf.dev.yugabyte.com/report/view/W3sibmFtZSI6InRlc3QtaWQiLCJpc0Jhc2VsaW5lIjp0cnVlLCJ0ZXN0X2lkIjo2ODI1MjAyfV0=
- Backward scan post updates: https://perf.dev.yugabyte.com/report/view/W3sibmFtZSI6InRlc3QtaWQiLCJpc0Jhc2VsaW5lIjp0cnVlLCJ0ZXN0X2lkIjo2ODk5ODAyfV0=

### CODE REVIEW

File: config/yugabyte/regression_pipelines/orderby_workloads/yb_colocated/ORDG8_orderby_backwardscan_post_updates.yaml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+#Orderby clause on range secondary indexed column of bigint datatype
+type: YUGABYTE
+driver: com.yugabyte.Driver
+url: jdbc:yugabytedb://{{endpoint}}:5433/yugabyte?sslmode=require&ApplicationName=featurebench&reWriteBatchedInserts=true&load-balance=false
+createdb: drop database if exists yb_colocated; create database yb_colocated with colocated=true
+username: {{username}}
+password: {{password}}
+batchsize: 128
+isolation: TRANSACTION_REPEATABLE_READ
+loaderthreads: 1
```
Comment (shaharuk-yb) [inline config/yugabyte/regression_pipelines/orderby_workloads/yb_colocated/ORDG8_orderby_backwardscan_post_updates.yaml:18]: loaderThreads?

File: config/yugabyte/regression_pipelines/orderby_workloads/postgres/ORDG8_orderby_backwardscan_post_updates.yaml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+#Orderby clause on hash secondary indexed column of bigint datatype
+#Orderby clause on range secondary indexed column of bigint datatype
+type: POSTGRES
+driver: org.postgresql.Driver
+url: jdbc:postgresql://{{endpoint}}:5432/postgres?sslmode=require
+username: {{username}}
+password: {{password}}
+batchsize: 128
+isolation: TRANSACTION_REPEATABLE_READ
+loaderthreads: 1
```
Comment (shaharuk-yb) [inline config/yugabyte/regression_pipelines/orderby_workloads/postgres/ORDG8_orderby_backwardscan_post_updates.yaml:17]: loaderThreads?

File: config/yugabyte/regression_pipelines/conditional_workloads/yugabyte/CW3_select_for_update.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+type: YUGABYTE
+driver: com.yugabyte.Driver
+url: jdbc:yugabytedb://{{endpoint}}:5433/yugabyte?sslmode=require&ApplicationName=featurebench&reWriteBatchedInserts=true&load-balance=false
+username: {{username}}
+password: {{password}}
+batchsize: 128
+isolation: TRANSACTION_REPEATABLE_READ
+loaderthreads: 1
```
Comment (shaharuk-yb) [inline config/yugabyte/regression_pipelines/conditional_workloads/yugabyte/CW3_select_for_update.yaml:8]: loaderThreads?

File: config/yugabyte/regression_pipelines/orderby_workloads/yb_colocated/ORDG8_orderby_backwardscan_post_updates.yaml (hunk 1 above)
Comment (shantanugupta-yb) [reply config/yugabyte/regression_pipelines/orderby_workloads/yb_colocated/ORDG8_orderby_backwardscan_post_updates.yaml]: Fixed. Though I would prefer that framework to by default load data in parallel when multiple independent tables are part of YAML unless there is a dependency between tables, like Foreign keys, which is when we load tables serially in a sequence using single thread.

File: config/yugabyte/regression_pipelines/orderby_workloads/postgres/ORDG8_orderby_backwardscan_post_updates.yaml (hunk 2 above)
Comment (shantanugupta-yb) [reply config/yugabyte/regression_pipelines/orderby_workloads/postgres/ORDG8_orderby_backwardscan_post_updates.yaml]: Fixed

File: config/yugabyte/regression_pipelines/conditional_workloads/yugabyte/CW3_select_for_update.yaml (hunk 3 above)
Comment (shantanugupta-yb) [reply config/yugabyte/regression_pipelines/conditional_workloads/yugabyte/CW3_select_for_update.yaml]: Fixed

**Pull request merged**